### PR TITLE
feat: add remaining steps for wallet transfer

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description
---
Add remaining steps in the `WalletTransfer.feature` file.

Motivation and Context
---

How Has This Been Tested?
---

### CODE REVIEW

File: integration_tests/tests/cucumber.rs [hunk 1]
```diff
@@ -400,24 +416,44 @@ async fn prune_node_connected_to_base_node(
...
+        .address
+        .to_hex();
+    let tx_ids = world.wallet_tx_ids.get(&wallet_address).unwrap();
 
     let num_retries = 100;
 
     for tx_id in tx_ids {
         println!("waiting for tx with tx_id = {} to be mined_confirmed", tx_id);
-        'inner: for _ in 0..num_retries {
+        for retry in 0..=num_retries {
```
Comment (SWvheerden) [inline integration_tests/tests/cucumber.rs:438]: nit: Technically this counts higher than the num of retries by 1

File: integration_tests/tests/cucumber.rs [hunk 2]
```diff
@@ -641,14 +681,17 @@ async fn send_amount_from_source_wallet_to_dest_wallet_without_broadcast(
     let tx_id = tx_res.transaction_id;
 
     // insert tx_id's to the corresponding world mapping
-    let mut source_tx_ids = world.wallet_tx_ids.get(&source_wallet_pubkey).unwrap().clone();
-    let mut dest_tx_ids = world.wallet_tx_ids.get(&dest_wallet_pubkey).unwrap().clone();
+    // insert tx_id's to the corresponding world mapping
```
Comment (SWvheerden) [inline integration_tests/tests/cucumber.rs]: double comment?

File: integration_tests/tests/cucumber.rs [hunk 3]
```diff
@@ -754,14 +796,17 @@ async fn send_one_sided_transaction_from_source_wallet_to_dest_wallt(
     }
 
     // insert tx_id's to the corresponding world mapping
-    let mut source_tx_ids = world.wallet_tx_ids.get(&source_wallet_pubkey).unwrap().clone();
-    let mut dest_tx_ids = world.wallet_tx_ids.get(&dest_wallet_pubkey).unwrap().clone();
+    // insert tx_id's to the corresponding world mapping
```
Comment (SWvheerden) [inline integration_tests/tests/cucumber.rs]: double comment

File: integration_tests/tests/cucumber.rs [hunk 4]
```diff
@@ -68,6 +70,8 @@ use crate::utils::{
 
 pub const LOG_TARGET: &str = "cucumber";
 pub const LOG_TARGET_STDOUT: &str = "stdout";
+const BLOCK_REWARD: u64 = 5_000_000;
```
Comment (SWvheerden) [inline integration_tests/tests/cucumber.rs]: this does not seem correct?
How did you calculate this?
This is supposed to be a decaying nonconstant number + varying fee

File: integration_tests/tests/cucumber.rs [hunk 5]
```diff
@@ -1509,7 +1642,518 @@ async fn transfer_tari_from_wallet_to_receiver(world: &mut TariWorld, amount: u6
...
+    base_node: String,
+) {
+    let peer_seeds = world.base_nodes.get(&base_node).unwrap().seed_nodes.clone();
+    println!(
+        "Start a new wallet {} connected to base node {}",
+        wallet.as_str(),
+        base_node.as_str()
+    );
+    spawn_wallet(world, wallet.clone(), Some(base_node.clone()), peer_seeds, None).await;
+    let num_blocks = amount * 1_000_000 / BLOCK_REWARD;
```
Comment (SWvheerden) [inline integration_tests/tests/cucumber.rs]: you can get the actual block reward from the consensus constants code directly?

File: integration_tests/tests/utils/mod.rs [hunk 6]
```diff
@@ -32,6 +32,8 @@ pub mod wallet;
 pub mod wallet_client;
 pub mod wallet_process;
 
+use tokio::time::Instant;
```
Comment (SWvheerden) [inline integration_tests/tests/utils/mod.rs]: why this change?
Did it not compile before

File: integration_tests/tests/cucumber.rs (hunk 1 above)
Comment (jorgeantonio21) [reply integration_tests/tests/cucumber.rs:438]: The idea is to have a failure check `retry`, but can make it consistent with the other methods, as well.

File: integration_tests/tests/cucumber.rs (hunk 3 above)
Comment (jorgeantonio21) [reply integration_tests/tests/cucumber.rs]: Removed

File: integration_tests/tests/cucumber.rs (hunk 4 above)
Comment (jorgeantonio21) [reply integration_tests/tests/cucumber.rs]: I did get it from the JS version. Take a look at `integration_tests/helpers/constants.js`

File: integration_tests/tests/cucumber.rs (hunk 2 above)
Comment (jorgeantonio21) [reply integration_tests/tests/cucumber.rs]: Removed double comments

File: integration_tests/tests/utils/mod.rs (hunk 6 above)
Comment (jorgeantonio21) [reply integration_tests/tests/utils/mod.rs]: Artifacts from a previous version

File: integration_tests/tests/cucumber.rs (hunk 4 above)
Comment (SWvheerden) [reply integration_tests/tests/cucumber.rs]: There is a very good chance this might only work under very strict circumstances. The JS code was old, and not always maintained. Perhaps see if you can calculate it some other way, perhaps using the constants directly. Using RUST we now have access to that code. 

Although this is still a strange way of calculating the number of blocks mined.

Comment (jorgeantonio21) [reply integration_tests/tests/cucumber.rs]: Refactored code to use ConsensusConstants directly and compute block reward from height

**Comment by SWvheerden** [comment]
LGTM

**Review by brianp — CHANGES_REQUESTED** [review]
Only request would be to remove the changes from `integration_tests/features/WalletTransfer.feature`

They don't hurt anything really, but they're also unnecessary changes.
